app/helpers/patients_helper.rb

```diff
@@ -28,6 +28,17 @@ def language_options
     standard_options + Config.find_or_create_by(config_key: 'language').options
   end
 
+  # TODO: do we want to do something similar to insurance here, to add new
+  # options to list if not already?
+  def voicemail_options
```

Comment:
@elimbaum , how much of a hassle would it be to add the something-similar-to-insurance to this branch? If we don't work this out then the behavior on delete becomes 'the next time the patient's record gets saved, VM preferences becomes No ID VM' which, given the sensitivity of this field for callers, exposes people to a little bit of risk. I don't think the code needs to be dry at this juncture (since, as you've noted, there's a clear next step best handled as a separate matter) but I do think we should sort this out before we ship. What do you think?

Reply:
I was thinking that (since this is a common issue across many fields) we should rewrite `options_for_select` to exhibit this behavior for all fields within the patient information page. I'm not sure it makes sense to only copy it over to voicemail, unless you think there's a reason we wouldn't want that behavior for other fields (language, insurance, etc.).

@lomky pointed out that we could run in to a situation where someone mispells a config. We risk losing all users' configs if a correction to the config is made, but users aren't updated immediately.

If you think that makes sense, I was going to open an issue and work on that next.

Reply:
I think this applies to:
- external pledge source
- language
- insurance
- support
- limit help text
- (and now) voicemail

Reply:
I think the options_for_select approach is a good idea as a next up, and that list of fields looks right to me at a glance (excepting limit help text, which I believe is an FYI and not something that gets saved on patient).

in the interim, could we copy the way `insurance_options` handles this? e.g. something like (pseudocoding here):

```
def voicemail_options(current_value = nil)
  standard_options = [...]
  custom_options = Config.find_by(config_key: 'voicemail_options')
  full_set = standard_options + custom_options

  if current_value.present? && ...
    full_set.push(current_value)
  end
  full_set
end
```

I think that will hit the goal of a) not deleting potentially sensitive entries from voicemail and b) keeping the door open to implement the more well-rounded and practical/safe approach you're advocating for here.

I think in the 'misspells a config' case @lomky identified, we'd be covered by the (misspelled) contents staying around with the current_value thing above, so we're safe on that front.

Does that seem like it makes sense?